src/server/services/wfs/qgswfstransaction.cpp

```diff
@@ -737,9 +772,10 @@ namespace QgsWfs
       action.error = false;
 
       // Get the Feature Ids of the inserted feature
-      for ( int j = 0; j < featureList.size(); j++ )
+      QgsAttributeList pkAttributes = provider->pkAttributeIndexes();
+      for ( const QgsFeature feat : featureList )
```

Comment:
I think it should be `for ( const QgsFeature &feat : qgis::as_const( featureList ) )`

Reply:
```suggestion
      for ( const QgsFeature &feat : qgis::as_const( featureList ) )
```